visibility-sensor.js

```diff
@@ -93,6 +83,33 @@ module.exports = React.createClass({
     return this.props.containment || window;
   },
 
+  addEventListenerWithDebounce: function (target, event, delay) {
+    if (!this.debounceCheck) {
+      this.debounceCheck = {};
+    }
+
+    var timeout;
+
+    var debounce = function() {
+      var context = this, args = arguments;
+      var later = function() {
+        timeout = null;
+        context.check.apply(context, args);
+      };
+      clearTimeout(timeout);
+      timeout = setTimeout(later, delay || 0);
```

Comment:
We need `this.lastTimeout = timeout;` back under `timeout = setTimeout(later, delay || 0);`